File: python/src/coordinate.py
class Coordinate:
    _rank: int
    _file: int

    def __init__(self, r: int, f: int) -> None:
        self._rank = r
        self._file = f

    def __str__(self) -> None:
        return "(" + str(self._file) + ", " + str(self._rank) + ")"

    def __eq__(self, coord):
        return self._rank == coord._rank and self._file == coord._file

    def get_rank(self) -> int:
        return self._rank

    def get_file(self) -> int:
        return self._file

    def to_syntax(self) -> str:
        pass

File: cpp/src/coordinate.cpp
#include "coordinate.h"

#include "util.h"

bool operator==(const Coordinate &c1, const Coordinate &c2) {
  return c1.get_rank() == c2.get_rank() && c1.get_file() == c2.get_file();
}

std::ostream &operator<<(std::ostream &os, const Coordinate &c) {
  os << coordinate_to_uci(c);
  return os;
}